### ISSUE TITLE
False positive regression in v1.1.391 (sometimes)

### ISSUE THREAD

**Issue by Joelius300** [microsoft/pyright#9685]
**Describe the bug**

Python 3.13.1

When upgrading to v1.1.391, I suddenly get false positives. These did not show up on v1.1.390.

Could be related to #9606 but I was not able to verify that anything is corrupted on my machine. However, I don't get any errors inside site-packages or the stdlib, it's only checking my code.

Edit: I found the bug when working with Pydantic classes and searched for a cause there for a long time, but actually it's much simpler to trigger (see comment below).

setup env

```bash
conda create -n pyright-test
conda activate pyright-test
conda install pydantic pyright==1.1.391
```

create main.py:

```python
from pydantic import BaseModel
from datetime import datetime

class Question(BaseModel):
    question: str

class Answer(BaseModel):
    answer: str

class Payload(BaseModel):
    question: Question
    answer: Answer
    timestamp: datetime


p = Payload(
    question=Question(question="my question"),
    answer=Answer(answer="my answer"),
    timestamp=datetime.now(),
)

print(p)
```

Run pyright:

```
/home/X/projects/pyright-test/main.py
  /home/X/projects/pyright-test/main.py:17:14 - error: Argument of type "Question" cannot be assigned to parameter "question" of type "Any" in function "__init__"
    "Question" is not assignable to "Any" (reportArgumentType)
  /home/X/projects/pyright-test/main.py:18:12 - error: Argument of type "Answer" cannot be assigned to parameter "answer" of type "Any" in function "__init__"
    "Answer" is not assignable to "Any" (reportArgumentType)
  /home/X/projects/pyright-test/main.py:22:1 - error: Argument missing for parameter "flush" (reportCallIssue)
3 errors, 0 warnings, 0 informations
```

Expected result:

```
0 errors, 0 warnings, 0 informations
```

Verify it's a regression:

```
conda install pyright==1.1.390
```

Run it again and you get the expected result.

---

Pydantic Versions are:

```
pydantic                  2.10.3
pydantic-core             2.27.1
```

**Comment by Joelius300** [comment]
I should have worked my way up, not down, but this also triggers a false positive and is much simpler than anything involving Pydantic.

```python
print("hello world")
```

results in

```
/home/X/projects/pyright-test/main.py
  /home/X/projects/pyright-test/main.py:1:1 - error: Argument missing for parameter "flush" (reportCallIssue)
```

on 1.1.391 but not on 1.1.390.

Given how easy it is to trigger and how apparently broken everything is, I'm now seriously doubting that this is affecting everyone. Some more information on my environment.

- RHEL 9
- Linux 5.14.0-427.35.1.el9_4.x86_6
- Python 3.13.1

What else can I provide or should I try?

**Comment by erictraut** [comment]
I'm not able to repro the problem. Those errors indicate something is corrupt on your system — perhaps the install of 1.1.391. Ensure that you have sufficient disk space in your environment. Perhaps the install was unable to unpack all of the bundled stubs that ship with pyright.

If you aren't able to diagnose the problem, then my recommendation is to wait until the next release to see if that clears things up for you.

I'm going to close the issue because I can't think of anything actionable.
